scanpointgenerator/generators/powertermgenerator.py

```diff
@@ -0,0 +1,84 @@
+from scanpointgenerator.core import Generator
+from scanpointgenerator.compat import np
+
+def to_list(value):
+    if isinstance(value, list):
+        return value
+    else:
+        return [value]
+
+
+@Generator.register_subclass("scanpointgenerator:generator/PowerTermGenerator:1.0")
+class PowerTermGenerator(Generator):
+    """
+    Generate a line of points according the function
+        y = ((x-x0)/a)**n + y0
+    """
+
+    def __init__(self, axis, units, start, stop, exponent, divisor, offset):
+        """
+        y = ((x-x0)/a)**n + y0
+        Args:
+            axis: The scannable axis e.g. "dcm_energy"
+            units: The scannable units e.g. "keV"
+            start: The first position to be generated. Will determine x0 in the above equation
+            stop: Will determine scan size. The final generated position will not necessarily be this...
+            exponent: n in above equation
+            divisor: a in the above equation
+            offset: y0 in the above equation. Point of interest which will be most finely sampled
+                    e.g. 7.112 (for Fe K edge)
+        """
+
+        if start >= stop:
+            raise ValueError("Start value must be lower than stop value")
+
+        if exponent < 1 or exponent % 2 == 0:
+            raise ValueError("Exponent must be odd and positive integer")
```

Comment:
Is this _generally_ sensible? The function makes sense for even, even if the path goes back on itself.
It may make reasoning about start vs stop more tricky - perhaps the input needs reconsidering in that case?

Reply:
Forbidding fractional and negative powers is reasonable though - they don't play nice.

Reply:
Thanks for looking at this. I'm thinking of a particular use case, for which even exponents don't really make sense.
But you are right, this is a PowerTermGenerator, not an OddPowerTermGenerator. I guess I can allow even powers and ensure start >= offset, but it would be arbitrary which real root i'd pick for x0... maybe the positive one to prevent repeated positions?

>> Forbidding fraction...
Oops - not checking for this!

Reply:
@xen-0 Even exponents now supported.